src/client/Microsoft.Identity.Client/IAccount.cs

```diff
@@ -33,6 +34,10 @@ public interface IAccount
         /// </summary>
         /// <remarks>Can be null, for example if this account was migrated to MSAL.NET from ADAL.NET v3's token cache</remarks>
         AccountId HomeAccountId { get; }
-       
-   }
+
+        /// <summary>
+        /// Returns the authority from which the token is acquired.
+        /// </summary>
+        string GetAuthority();    
```

Comment:
IAccount is a public interface, so you cannot change it, because it'll break ppl. Both Java and C# solve this problem with "default interface methods", but sadly we cannot use this feature in C# because it is only available on .NET Core. 

So there are 2 practical ways of solving this: 

1. (we won't do this, but just for reference) Create a new interface IAccount2 which inherits from IAccount and adds the new member. Make our code use IAccount2 everywhere. Customers relying on IAccount will not be broken because of polymorphism. 

2. Create an extension method `GetAuthority()`

```csharp
public static class AccountExtensions
{
    public static string GetAuthority(this IAccount account)
    {
          $"https://{account.Environment}/{account.HomeAccountId.TenantId}";
    }
}

```

Reply:
Note that extension methods are a C# concept which only works for methods. This is why you can't have a property Authority ...